### ISSUE TITLE
Inconsistent parsing of exponents

### ISSUE THREAD

**Issue by isuruf** [symengine/symengine#1361]
_From @xwan242 on November 5, 2017 16:14_

Inserting a + or - in front of an exponent causes all multiplicative terms after the exponent to be grouped together, instead of the behaviour in standard Python and SymPy.

E.g.
```
>>> 3.0**-2*3
0.3333333333333333
>>> sympy.sympify('3.0**-2*3')
0.333333333333333
>>> symengine.sympify('3.0**-2*3') # = symengine.sympify('3.0**-6')
0.00137174211248285
```

_Copied from original issue: symengine/symengine.py#207_

**Comment by isuruf** [comment]
cc @srajangarg

**Comment by srajangarg** [comment]
What does `sympify` call internally? Just `parse`?

**Comment by isuruf** [comment]
Yes, just parse

**Comment by isuruf** [comment]
Fixed in #1362. Thanks @srajangarg
